demisto_sdk/scripts/tests/validate_mypy_global_ignore_test.py

```diff
@@ -0,0 +1,510 @@
+
+        from demisto_sdk.commands.common.git_util import Repo as GitRepo
+        from demisto_sdk.scripts.validate_deleted_files import GitUtil
+
+        mocker.patch.dict(os.environ, {"DEMISTO_SDK_CONTENT_PATH": git_repo.path})
+        mocker.patch.object(GitRepo, "remote", return_value="")
+        mocker.patch.object(GitUtil, "fetch", return_value=None)
+
+        # Initialize Pack
+        git_repo.create_pack(name="TestPack").create_integration(name="TestIntegration")
+        git_repo.git_util.commit_files("Added a new Pack and Integration")
+
+    def test_mypy_ignore_not_added(cls, git_repo: Repo):
+        """
+        Test the behavior of `validate_mypy_global_ignore` when
+        non-mypy ignore comment is added.
+
+        Given:
+        - A content git repo with an integration pushed to remote.
+        - A feature branch.
+
+        When:
+        - Valid Python code is added to the intengration code.
+
+        Then:
+        - Exit code 0 is returned.
+        """
+
+        git_repo.git_util.repo.git.checkout("-b", cls.branch_add_mypy_ignore)
+
+        py_file_path = Path(git_repo.packs[0].integrations[0].code.path)
+        py_file_path.write_text(cls.added_code)
+
+        git_repo.git_util.stage_file(py_file_path)
+
+        runner = CliRunner()
+
+        with ChangeCWD(git_repo.path):
+            result = runner.invoke(cls.func, [])
+
+        assert result.exit_code == 0
+
+    def test_mypy_ignore_add_not_global(cls, git_repo: Repo):
+        """
+        Test the behavior of `validate_mypy_global_ignore` when
+        a non-global mypy ignore comment is added to the end
+        of a statement.
+
+        Given:
+        - A content git repo with an integration pushed to remote.
+        - A feature branch.
+
+        When:
+        - Type ignore comment is added to the end of a print
+        statement.
+
+        Then:
+        - Exit code 0 is returned.
+        """
+
+        git_repo.git_util.repo.git.checkout("-b", cls.branch_add_mypy_ignore)
+
+        py_file_path = Path(git_repo.packs[0].integrations[0].code.path)
+        py_file_path.write_text(cls.added_mypy_ignore)
+
+        git_repo.git_util.stage_file(py_file_path)
+
+        runner = CliRunner()
+
+        with ChangeCWD(git_repo.path):
+            result = runner.invoke(cls.func, [])
+
+        assert result.exit_code == 0
+
+    def test_mypy_ignore_add_tab(cls, git_repo: Repo):
+        """
+        Test the behavior of `validate_mypy_global_ignore` when
+        a non-global mypy ignore comment is added in the beginning
+        of the statement with a tab.
+
+        Given:
+        - A content git repo with an integration pushed to remote.
+        - A feature branch.
+
+        When:
+        - Type ignore comment is added after a tab.
+
+        Then:
+        - Exit code 0 is returned.
+        """
+
+        git_repo.git_util.repo.git.checkout("-b", cls.branch_add_mypy_ignore)
```

Comment:
do we actually checkout during a UT? Sounds unhealthy

Reply:
Removed as part of removal of `git` usage to find changed files https://github.com/demisto/demisto-sdk/pull/4405/commits/24b66fd5ba7781ebe5267ea8cc5004a29f9b21a1